swaybar/ipc.c

```diff
@@ -298,6 +304,23 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 					calloc(1, sizeof(struct swaybar_workspace));
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
+				ws->label = strdup(ws->name);
+				char *ws_dup = strdup(ws->name);
+				char *stripped_name = strtok(ws_dup, ":");
```

Comment:
I just did some testing and it seems like i3 strips names even if the colon is not present: i.e. "1a" will be stripped into 1 and a. So this logic will have to be changed if we want to match their behaviour.

Reply:
I think this could be a bug. The [docs](https://i3wm.org/docs/userguide.html#_strip_workspace_numbers_name) for `strip_workspace_name` and `strip_workspace_number` emphasise that the name needs to be in the form `[n]:[NAME]` to apply.

Reply:
They have confirmed that this is intended. I will see if I can get this done soon.